test/unittests/execute_numeric_test.cpp

```diff
@@ -3,22 +3,28 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "execute.hpp"
+#include "instructions.hpp"
 #include "parser.hpp"
 #include <gtest/gtest.h>
 #include <test/utils/asserts.hpp>
 #include <test/utils/execute_helpers.hpp>
 #include <test/utils/hex.hpp>
+#include <test/utils/typed_value.hpp>
 
 using namespace fizzy;
 using namespace fizzy::test;
 
 namespace
 {
-ExecutionResult execute_unary_operation(Instr instr, uint64_t arg)
+ExecutionResult execute_unary_operation(Instr instr, TypedValue arg)
 {
+    const auto& instr_type = get_instruction_type_table()[static_cast<size_t>(instr)];
+    EXPECT_EQ(instr_type.inputs.size(), 1);
```

Comment:
size checks should be better `ASSERT`s, I think

Reply:
`ASSERT`s does not work here.

Reply:
`ASSERT` returns void. It should be possible to use an `ASSERT` if converted to a proper test fixture. But I don't want to mix such changes. Refactoring is in #651.